app.js

```diff
@@ -1,249 +1,34 @@
-      );
-      app.use(
-        "/groups",
-        require("./server_src/routes/printerGroups", { page: "route" })
-      );
-      app.use(
-        "/settings",
-        require("./server_src/routes/settings", { page: "route" })
-      );
-      app.use(
-        "/printersInfo",
-        require("./server_src/routes/SSE-printersInfo", { page: "route" })
-      );
-      app.use(
-        "/dashboardInfo",
-        require("./server_src/routes/SSE-dashboard", { page: "route" })
-      );
-      app.use(
-        "/monitoringInfo",
-        require("./server_src/routes/SSE-monitoring", { page: "route" })
-      );
-      app.use(
-        "/filament",
-        require("./server_src/routes/filament", { page: "route" })
-      );
-      app.use(
-        "/history",
-        require("./server_src/routes/history", { page: "route" })
-      );
-      app.use(
-        "/scripts",
-        require("./server_src/routes/scripts", { page: "route" })
-      );
-      app.use(
-        "/input",
-        require("./server_src/routes/externalDataCollection", { page: "route" })
-      );
-      app.use(
-        "/client",
-        require("./server_src/routes/sorting", { page: "route" })
-      );
-      app.get("*", function(req, res) {
-        res.redirect("/");
-      });
-    } catch (e) {
-      await logger.error(e);
-    }
-  }
-};
-
-// Mongo Connect
 mongoose
-  .connect(dbConnectionString, {
+  .connect(process.env.MONGO, {
     useNewUrlParser: true,
     useUnifiedTopology: true,
-    useFindAndModify: false
-  })
-  .then(async () => {
-    //Check database actually works...
-    if (mongoose.connection.readyState !== 1) {
-      await databaseIssue();
-      let err = "No db connection...";
-      throw err;
-    }
+    useFindAndModify: false,
+    serverSelectionTimeoutMS: 2500,
   })
-  .then(() => setupServerSettings())
-  .then(() => serverStart())
+  .then(() => ensureSystemSettingsInitiated())
+  .then(() => serveOctoFarmNormally(octoFarmServer))
   .catch(async (err) => {
-    console.error("\n\n\nDATABASE ISSUE!!!");
     logger.error(err);
-    await databaseIssue();
+    serveDatabaseIssueFallback(octoFarmServer);
   });
 
-const databaseIssue = async () => {
-  app.listen(4000, "0.0.0.0", () => {
-    logger.info("HTTP server started...");
-    logger.info(`You have database issues... web interface loaded on: ${4000}`);
-    console.log(`You have database issues... web interface loaded on: ${4000}`);
-    // This only works when this is a child process (like when managed by PM2 f.e.)
-    if (typeof process.send === "function") {
-      // eslint-disable-next-line no-console
-      process.send("ready");
-    }
-  });
-
-  app.use("/", require("./server_src/routes/databaseIssue", { page: "route" }));
-  app.use("/serverChecks", require("./server_src/routes/serverChecks", { page: "route" }));
-  app.get("*", function(req, res) {
-    res.redirect("/");
-  });
-};
+bootAutoDiscovery();
```

Comment:
Asynchronous, but **after** mongoose and **after** the API has been triggered to start.

This was probably the reason why OF was that slow to boot (and the cleaners)